boto3/dynamodb/transform.py

```diff
@@ -0,0 +1,188 @@
+# Copyright 2015 Amazon.com, Inc. or its affiliates. All Rights Reserved.
+#
+# Licensed under the Apache License, Version 2.0 (the "License"). You
+# may not use this file except in compliance with the License. A copy of
+# the License is located at
+#
+# http://aws.amazon.com/apache2.0/
+#
+# or in the "license" file accompanying this file. This file is
+# distributed on an "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF
+# ANY KIND, either express or implied. See the License for the specific
+# language governing permissions and limitations under the License.
+from collections import Mapping
+import functools
+
+from boto3.dynamodb.types import TypeSerializer, TypeDeserializer
+from boto3.dynamodb.conditions import ConditionBase
+from boto3.dynamodb.conditions import ConditionExpressionBuilder
+
+
+def register_high_level_interface(base_classes, **kwargs):
+    # This assumes the base class is the base resource object
+    resource_base_class = base_classes[0]
+
+    class DynamoDBHighLevelResource(resource_base_class):
```

Comment:
I was actually suggesting just making it a mixin class that was also a base class for the passed in resource class, i.e

```
class DDBHLResource(object):
   def __init__(self, ...): # all the same code you have

def register_...():
  base_classes.insert(0, DDBHLResource)

```

That should avoid the problem of subclass from `ServiceResource` you mention, and still work with the event emitter requirements you have wouldn't it?